src/impl/android/external_commit_helper.cpp

```diff
@@ -89,24 +89,43 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
     int ret = mkfifo(path.c_str(), 0600);
     if (ret == -1) {
         int err = errno;
-        if (err == ENOTSUP) {
+        if (err == ENOTSUP || err == EACCES) {
             // Filesystem doesn't support named pipes, so try putting it in tmp instead
             // Hash collisions are okay here because they just result in doing
             // extra work, as opposed to correctness problems
-            std::ostringstream ss;
-
-            std::string tmp_dir(getenv("TMPDIR"));
-            ss << tmp_dir;
-            if (tmp_dir.back() != '/')
-              ss << '/';
-            ss << "realm_" << std::hash<std::string>()(path) << ".note";
-            path = ss.str();
-            ret = mkfifo(path.c_str(), 0600);
-            err = errno;
+
+            std::string tmp_dir = parent.get_temp_dir();
+            if (tmp_dir.empty()) {
+                auto tmp_dir_from_env = getenv("TMPDIR");
+                if (tmp_dir_from_env) {
+                    tmp_dir = std::string(tmp_dir_from_env);
+                }
+            }
+            if (!tmp_dir.empty()) {
+                std::ostringstream ss;
+                ss << tmp_dir;
+                if (tmp_dir.back() != '/')
```

Comment:
Right, but this helper class is for android only and all android uses / as a separator.